service/api_logic/games_logic.py

```diff
@@ -3,63 +3,28 @@
 from database.models import Games, Country, TeamIndex, League, Sport
 from dto.pagination import Pagination
 from exept.handle_exeptions import handle_exceptions
-from service.api_logic.scripts import apply_filters
 from sqlalchemy.orm import aliased
-from database.session import SessionLocal
 from logger.logger import Logger
+from service.api_logic.filter_manager.filter_manager_factory import FilterManagerFactory
 
-logger = Logger("logger", "all.log")
+class GamesService:
+    def __init__(self, games_dal):
+        self._games_dal = games_dal
+        self._logger = Logger("logger", "all.log").logger
 
-session = SessionLocal()
+    def get_games_today(self, filters_dto: GamesDTO(), pagination: Pagination):
 
-@handle_exceptions
-@logger.log_function_call()
-def get_games_today(
-        filters_dto: dict,
-        pagination: Pagination
-):
-    home_team = aliased(TeamIndex)
-    away_team = aliased(TeamIndex)
+        query = self._games_dal.get_query()
 
-    query = (
-        session.query(
-            Games,
-            League.name.label("league_name"),
-            League.logo.label("league_logo"),
-            Country.name.label("country_name"),
-            home_team.name.label("home_team_name"),
-            home_team.logo.label("home_team_logo"),
-            away_team.name.label("away_team_name"),
-            away_team.logo.label("away_team_logo"),
-            Games.score_home_team,
-            Games.score_away_team,
-            Games.status,
-            Games.time,
-            Games.date,
-            Games.api_id
-        )
-        .join(League, Games.league_id == League.league_id)
-        .join(Country, Games.country_id == Country.country_id)
-        .join(Sport, Games.sport_id == Sport.sport_id)
-        .join(home_team, Games.team_home_id == home_team.team_index_id)
-        .join(away_team, Games.team_away_id == away_team.team_index_id)
-    )
+        filtered_query = FilterManagerFactory.apply_filters(Games, query, filters_dto)
 
-    model_aliases = {
-        "games": Games,
-        "countries": Country,
-        "leagues": League,
-    }
+        offset, limit = pagination.get_pagination()
+        if offset is not None and limit is not None:
+            filtered_query = filtered_query.offset(offset).limit(limit)
 
-    query = apply_filters(query, filters_dto, model_aliases)
+        games = self._games_dal.execute_query(filtered_query)
 
-    offset, limit = pagination.get_pagination()
-    if offset is not None and limit is not None:
-        query = query.offset(offset).limit(limit)
-
-    games = query.all()
-
-    schema = GameOutput(many=True)
-    return schema.dump(games)
+        schema = GameOutput(many=True)
```

Comment:
I hate it so much when you guys name a variable "schema" when it has no business having that name whatsoever